messaging/src/main/java/io/lantern/messaging/Messaging.kt

```diff
@@ -343,13 +346,21 @@ class Messaging(
         }
         replyToSenderId?.let { msgBuilder.setReplyToSenderId(it) }
         replyToId?.let { msgBuilder.setReplyToId(it) }
-        val out =
-            Model.OutboundMessage.newBuilder().setMessageId(base32Id)
+        var out: Model.OutboundMessage.Builder? = null
+        if(recipientId != myId.id){
+            out = Model.OutboundMessage.newBuilder().setMessageId(base32Id)
                 .setSent(sent)
                 .setSenderId(myId.id)
                 .setRecipientId(recipientId)
+        }else{
+            msgBuilder.status = Model.StoredMessage.DeliveryStatus.COMPLETELY_SENT
+            msgBuilder.direction = Model.MessageDirection.IN
+        }
         var attachmentId = 0
         attachments?.forEach { attachment ->
+            if(recipientId == myId.id){
```

Comment:
Since this condition is repeated, you might want to evaluate it only once and assign it to a val like `sendingToSelf`.

Reply:
Definitely 👍